plugins/inputs/logparser/logparser.go

```diff
@@ -82,7 +82,8 @@ func (l *LogParserPlugin) Gather(acc telegraf.Accumulator) error {
 	defer l.Unlock()
 
 	// always start from the beginning of files that appear while we're running
-	return l.tailNewfiles(true)
+	// why need to set true? It's ok to set l.FromBeginning.
+	return l.tailNewfiles(l.FromBeginning)
```

Comment:
When this parameter is false, we seek to the end of the file before reading.  To ensure the entire file is read we need to leave this as true.